src/socket.c

```diff
@@ -1446,8 +1446,12 @@ static void setupnextconn(Sock *sock)
 {
     struct addrinfo *ai, *next = sock->addr;
 
-    if (sock->fd >= 0)
-	close(sock->fd);
+    if (sock->fd >= 0) {
+        FD_CLR(sock->fd, &readers);
+        FD_CLR(sock->fd, &writers);
+	      close(sock->fd); 
```

Comment:
Whitespace issues?

Reply:
Good catch, Github's web-merge is apparently something to be careful with in that regard. Just pushed corrections.